nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/test/java/org/apache/nifi/processors/media/TestExtractMediaMetadata.java

```diff
@@ -0,0 +1,411 @@
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        flowFile0 = successFiles.get(0);
+        flowFile0.assertAttributeExists("filename");
+        flowFile0.assertAttributeExists("txt.X-Parsed-By");
+    }
+
+    @Test
+    public void testMaxAttributes() throws IOException {
+        TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/notImage.txt").toPath());
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        List<MockFlowFile> successFiles0 = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile0 = successFiles0.get(0);
+        int fileAttrCount0 = 0;
+        for (Map.Entry attr : flowFile0.getAttributes().entrySet()) {
+            if (attr.getKey().toString().startsWith("txt.")) {
+                fileAttrCount0++;
+            }
+        }
+        assertTrue(fileAttrCount0 > 1);
+
+        runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.MAX_NUMBER_OF_ATTRIBUTES, Integer.toString(fileAttrCount0 - 1));
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/notImage.txt").toPath());
+        runner.run();
+
+        runner.assertAllFlowFilesTransferred(ExtractMediaMetadata.SUCCESS, 1);
+        runner.assertTransferCount(ExtractMediaMetadata.FAILURE, 0);
+
+        List<MockFlowFile> successFiles = runner.getFlowFilesForRelationship(ExtractMediaMetadata.SUCCESS);
+        MockFlowFile flowFile1 = successFiles.get(0);
+        int fileAttrCount1 = 0;
+        for (Map.Entry attr : flowFile1.getAttributes().entrySet()) {
+            if (attr.getKey().toString().startsWith("txt.")) {
+                fileAttrCount1++;
+            }
+        }
+        assertEquals(fileAttrCount0, fileAttrCount1 + 1);
+    }
+
+    @Test
+    public void testBmp() throws IOException {
+        final TestRunner runner = TestRunners.newTestRunner(new ExtractMediaMetadata());
+        runner.setProperty(ExtractMediaMetadata.FILENAME_FILTER, ".*[tTxXtT]");
+        runner.setProperty(ExtractMediaMetadata.MIME_TYPE_FILTER, "text/.*");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_FILTER, "");
+        runner.setProperty(ExtractMediaMetadata.METADATA_KEY_PREFIX, "txt.");
+        runner.assertValid();
+
+        runner.enqueue(new File("target/test-classes/notImage.txt").toPath());
```

Comment:
This test is titled "testBmp" but runs and verifies the same "notImage.txt" as prior tests use.